internal/storage/image.go

```diff
@@ -298,6 +306,31 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 	return results, nil
 }
 
+// determines if an images is intermediate, based on ID
+func isIntermediate(image *storage.Image, server ImageServer) (bool, error) {
```

Comment:
If you make this a part of `ImageService`, then you won't have to pass `server` as a param here.